src/components/article/CommentCard.jsx

```diff
@@ -1,15 +1,68 @@
 import "./Article.css";
-import  dayjs  from 'dayjs'
-import advancedFormat from 'dayjs/plugin/advancedFormat'
+import dayjs from "dayjs";
+import advancedFormat from "dayjs/plugin/advancedFormat";
+import { useContext, useState } from "react";
+import { MdDelete } from "react-icons/md";
+import { UsernameContext } from "../../contexts/Username";
+import { deleteComment } from "../api";
+
+export default function CommentCard({ comment, handleViewComments }) {
+  dayjs.extend(advancedFormat);
+  const { username } = useContext(UsernameContext);
+  const [isDeleting, setIsDeleting] = useState(false);
+  const [isDeleted, setIsDeleted] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  function handleDeleteComment() {
+    setIsDeleting(true);
+    deleteComment(comment.comment_id)
+      .then(() => {
+        setIsDeleting(false);
+        setIsDeleted(true);
+        handleViewComments();
+      })
+      .catch(() => {
+        setIsError(true);
+      });
+  }
```

Comment:
Great that you're letting the user know when the request hasn't gone through, and have a loading pattern for when it's in progress 👍